lightning/src/ln/outbound_payment.rs

```diff
@@ -1607,14 +1607,10 @@ impl OutboundPayments {
 
 	pub(super) fn add_new_awaiting_invoice(
 		&self, payment_id: PaymentId, expiration: StaleExpiration, retry_strategy: Retry,
-		max_total_routing_fee_msat: Option<u64>, retryable_invoice_request: Option<RetryableInvoiceRequest>
+		route_params_config: Option<RouteParametersConfig>, retryable_invoice_request: Option<RetryableInvoiceRequest>
 	) -> Result<(), ()> {
 		let mut pending_outbounds = self.pending_outbound_payments.lock().unwrap();
-		let route_params_config = max_total_routing_fee_msat.map_or(
-			RouteParametersConfig::new(),
-			|fee_msats| RouteParametersConfig::new()
-				.with_max_total_routing_fee_msat(fee_msats)
-		);
+		let route_params_config = route_params_config.unwrap_or(RouteParametersConfig::new());
```

Comment:
In this commit, we should also set `max_total_routing_fee_msat` in `AwaitingInvoice` to what is in the config in order to support downgrades.

Reply:
Done! Update the code in [pr3342.06](https://github.com/shaavan/rust-lightning/commits/pr3342.06) to write max_total_routing_fee_msat to support downgrades too!
Thanks for the suggestion!